### PR TITLE
docs: final draft notification pattern

### LINKED ISSUE #332: Notification pattern
**Abstract**
Similar to Shell component : Navigation pattern :: Form component : Form pattern we should create a supporting pattern for the notification component. Pattern guides users on when to use notifications and how to choose which notification component to use, rather than the notification component itself. 

**Link to preview:**
https://carbon-website-git-fork-connor-leech-2019-patterns-master.carbon-design-system.now.sh/patterns/notification-pattern/

**Development activity**
- [x] Peer playback, Nov 15
- [x] Early SME review (Maui Frances, Alex Swain), Nov 19
- [x]  Image requirements walkthrough (Lauren, Jeanie), Nov 21
- [x]  Chris Connors' outline review, Nov 21
- [x]  Content review (Jan), Nov 26
- [x] Accessibility review with D.A. Kahn, Nov 26
- [x] Dev review with Josh Black, Nov 26
- [x] Playback with Craig, Dec 2

- [x] SME reviews:
    - [x] Peter Vachon - feedback incorporated
    - [x] Alex Swain - feedback incorporated
    - [x] Maui Francis - feedback incorporated
    - [x] Diana Tran - feedback incorporated
    - [x] David Levinson - feedback incorporated 
    - [ ] Luke Firth, sent Nov 25
    - [ ] Jeff Placer, sent Dec 3
    - [ ] Christopher DeCicco, sent Dec 3

**Existing notification guidance:**
Cloud
https://pages.github.ibm.com/ibmcloud/pal/experimental/notification-usage/

IoT
https://pages.github.ibm.com/Watson-IoT/IoT-Design-Site/patterns/alert-configuration/overview/
https://pages.github.ibm.com/Watson-IoT/IoT-Design-Site/patterns/alert-console/overview/

Security
https://github.ibm.com/security/isc-patterns/tree/master/layouts/notifications

WCE
http://billboard1.fyre.ibm.com/component-library/messages/build
http://billboard1.fyre.ibm.com/component-library/message-banner
http://billboard1.fyre.ibm.com/component-library/modal-nag

Cloud data and AI
https://github.ibm.com/CloudIntegrationDesign/DesignSystemAdoptionGuild/issues/47
https://github.ibm.com/CloudIntegrationDesign/DesignSystemAdoptionGuild/issues/83

### PR DESCRIPTION
**Pull request by jillianhowarth**
Closes #332 

Moving notification pattern into new branch for publishing.

### CODE REVIEW

File: src/pages/patterns/notification-pattern/index.mdx [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+| Relevant      | Notifications should be related to the user's goals and presented in the context of what they are doing |
+| Timely        | Users should be kept up-to-date with prompt notifications and ensure they see critical immediately      |
+| Informative   | Provide users with the context and next-steps needed to understand and address the notification         |
+
+Be considerate of your users when sending a notification. Notifications that are too frequent or disruptive create negative experiences and drive users from platforms.
+
+Notifications are comprised of status and type. Their status signifies the purpose of the information being conveyed. Notification types allow you to tailor the disruptiveness of the notification to the specific situation. Notification status and type options in Carbon should be combined to create notifications that are relevant, timely, and informative for each use case.
+
+<Row>
+<Column colSm={2} colMd={2} colLg={3}>
```
Comment (alisonjoseph) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
<Column colSm={2} colMd={2} colLg={4}>
```

The text in the titles is wrapping at the large breakpoint (looks fine on mobile/tablet) so might work better as a 4 column. Not a huge deal either way though.

File: src/pages/patterns/notification-pattern/index.mdx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+#### Notification status
+
+- Informational <br/>
+- Success <br/>
+- Warning <br/>
+- Error <br/>
+
+</Column>
+<Column colSm={2} colMd={2} colLg={3}>
```
Comment (alisonjoseph) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
<Column colSm={2} colMd={2} colLg={4}>
```
switch here too if you switch above

File: src/pages/patterns/notification-pattern/index.mdx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+<Row>
+<Column colSm={2} colMd={2} colLg={2}>
+
+#### Components
+
+- [Notification](/components/notification/usage) <br/>
+- [Modal](/components/modal/usage) <br/>
+- [UI shell](/components/UI-shell-right-panel/usage) <br/>
+
```
Comment (alisonjoseph) [inline src/pages/patterns/notification-pattern/index.mdx]: You shouldn't need the break tags here
```
- [Notification](/components/notification/usage) 
- [Modal](/components/modal/usage)
- [UI shell](/components/UI-shell-right-panel/usage) 
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+</Column>
+<Column colSm={2} colMd={2} colLg={3}>
+
+#### Patterns
+
+- Forms <br/>
+- Dialogs <br/>
+- Error states (future) <br/>
```
Comment (alisonjoseph) [inline src/pages/patterns/notification-pattern/index.mdx]: Same here, you can remove br tags

File: src/pages/patterns/notification-pattern/index.mdx [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+<Row>
+<Column colSm={2} colMd={2} colLg={3}>
+
+#### Notification status
+
+- Informational <br/>
+- Success <br/>
+- Warning <br/>
+- Error <br/>
+
```
Comment (alisonjoseph) [inline src/pages/patterns/notification-pattern/index.mdx]: you can remove br tags

File: src/pages/patterns/notification-pattern/index.mdx [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+<Column colSm={2} colMd={2} colLg={2}>
+
+#### Components
+
+- [Notification](/components/notification/usage) <br/>
+- [Modal](/components/modal/usage) <br/>
+- [UI shell](/components/UI-shell-right-panel/usage) <br/>
+
+</Column>
+<Column colSm={2} colMd={2} colLg={3}>
```
Comment (alisonjoseph) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
<Column colSm={2} colMd={2} colLg={4}>
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+**Don't use notifications that dismiss on a timer for critical or emergency messages.** Some users with disabilities need more time to read or interact with messages and timed toasts may not provide sufficient time.
+[WCAG 2.1 success criterion 2.2.4 (AAA)](https://www.w3.org/WAI/WCAG21/Understanding/no-timing.html)
+
+**Users should be able to manage or limit noncritical notifications.** This gives users the control to reduce the number of distractions or disruptions they receive, which is particularly helpful for users with cognitive limitations.
+[WCAG 2.1 success criterion 2.2.3 (AAA)](https://www.w3.org/WAI/WCAG21/Understanding/interruptions.html)
+
+## Related components and patterns
+
+<Row>
+<Column colSm={2} colMd={2} colLg={2}>
```
Comment (alisonjoseph) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
<Column colSm={2} colMd={2} colLg={4}>
```

UI shell is wrapping and looks odd

File: src/pages/patterns/notification-pattern/index.mdx [hunk 8]
```diff
@@ -0,0 +1,307 @@
...
+
+## Overview
+
+It’s important to keep users informed and send notifications when there is an update or status change they should be aware of. Users should always be given appropriate and timely messages to help them understand whether they are moving towards their goal. Ensure your notifications are relevant, timely, and informative.
+
+| Key principle | Definition                                                                                              |
+| ------------- | ------------------------------------------------------------------------------------------------------- |
+| Relevant      | Notifications should be related to the user's goals and presented in the context of what they are doing |
+| Timely        | Users should be kept up-to-date with prompt notifications and ensure they see critical immediately      |
+| Informative   | Provide users with the context and next-steps needed to understand and address the notification         |
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
| Informative   | Provide users with the context and next steps needed to understand and address the notification         |
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+</AnchorLinks>
+
+## Overview
+
+It’s important to keep users informed and send notifications when there is an update or status change they should be aware of. Users should always be given appropriate and timely messages to help them understand whether they are moving towards their goal. Ensure your notifications are relevant, timely, and informative.
+
+| Key principle | Definition                                                                                              |
+| ------------- | ------------------------------------------------------------------------------------------------------- |
+| Relevant      | Notifications should be related to the user's goals and presented in the context of what they are doing |
+| Timely        | Users should be kept up-to-date with prompt notifications and ensure they see critical immediately      |
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
| Timely        | Ensure users are kept up to date with prompt notifications and see critical notifications immediately      |
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+
+## Notification status
+
+Notification status helps convey the information being communicated. These statuses correspond with a color and icon to provide a consistent, universal experience for users.
+
+![Example of notification status](/images/Status_1120.png)
+
+| Status        | Usage                                                                                                                        | Action                                                                                                    | Color  | Icon               |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------- | ------ | ------------------ |
+| Informational | Provide additional information to users that may not be tied to their current action or task                                 | Do not require immediate action and can be dismiss on a timer or persist, depending on the content        | Blue   | No icon            |
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
| Informational | Provide additional information to users that may not be tied to their current action or task                                 | Do not require immediate action and can be dismissed on a timer or persist, depending on the content        | Blue   | No icon            |
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+| Success       | Confirm a task was completed as expected                                                                                     | Typically do not require further action and can dismiss automatically or persist in a nonintrusive manner | Green  | `checkmark filled` |
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
| Success       | Confirm a task was completed as expected                                                                                     | Typically do not require further action and can be dismissed automatically or persist in a nonintrusive manner | Green  | `checkmark filled` |
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+| Toast              | Short, time-based messages that slide in and out of a page and provide nondisruptive information                  | Disappear automatically or can be dismissed by user                                                                                             |
+| Banner             | System or product level notifications that are not specific to a task                                             | Persist until dismissed by user and may include a ghost button or link                                                                          |
+| Notification panel | Notification center that provides users with system-generated messages                                            | Opened and closed by user                                                                                                                       |
+| Modal              | Highly disruptive notifications that provide users with critical information that needs their attention or action | Persist and block tasks until dismissed by user. Modals allow more user action than other notifications and can include other Carbon components |
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
| Modal              | Highly disruptive notifications that provide users with critical information that needs their attention or action | Persist and block tasks until dismissed by user. Modals allow more user action than other notifications and can include other Carbon components. |
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 13]
```diff
@@ -0,0 +1,307 @@
...
+
+### Inline notification
+
+[Inline notifications](https://www.carbondesignsystem.com/components/notification/usage/) are noninvasive and confined to a specific area in the UI. Inline notifications display both task-generated and system-generated messages and persist until they are dismissed by the user or the notification is resolved. They are frequently used in conjunction with field-level messages for errors in forms or other input areas.
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
[Inline notifications](https://www.carbondesignsystem.com/components/notification/usage/) are nondisruptive and confined to a specific area in the UI. Inline notifications display both task-generated and system-generated messages and persist until they are dismissed by the user or the notification is resolved. They are frequently used in conjunction with field-level messages for errors in forms or other input areas.
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 14]
```diff
@@ -0,0 +1,307 @@
...
+<Column colLg={8}>
+
+![Notification panel in a product](/images/Panel_1120.png)
+
+</Column>
+</Row>
+
+### Modal
+
+Modals interrupt the user and pause their current task. They are highly disruptive to users and should be used sparingly. Only use a modal when the message is critical and needs the user's immediate attention or action. Modals persist until users engage with them and only disappear when dismiss by the user.
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
Modals interrupt the user and pause their current task. They are highly disruptive to users and should be used sparingly. Only use a modal when the message is critical and needs the user's immediate attention or action. Modals persist until users engage with them and only disappear when dismissed by the user.
```

File: src/pages/patterns/notification-pattern/index.mdx [hunk 15]
```diff
@@ -0,0 +1,307 @@
...
+
+#### Best practices:
+
+- Only display one modal notification at a time.
+- Use modals with messages that need to immediately interrupt a user's task.
```
Comment (janchild) [inline src/pages/patterns/notification-pattern/index.mdx]:
```suggestion
- Use modals when you need to immediately interrupt a user's task.
```

File: src/pages/patterns/notification-pattern/index.mdx (hunk 3 above)
Comment (jillianhowarth) [reply src/pages/patterns/notification-pattern/index.mdx]: Fixed!

File: src/pages/patterns/notification-pattern/index.mdx (hunk 4 above)
Comment (jillianhowarth) [reply src/pages/patterns/notification-pattern/index.mdx]: Fixed!

File: src/pages/patterns/notification-pattern/index.mdx (hunk 5 above)
Comment (jillianhowarth) [reply src/pages/patterns/notification-pattern/index.mdx]: Fixed!

**Pull request merged**